server/tests/test_login.py

```diff
@@ -0,0 +1,47 @@
+import json
+
+from server.models import Lecturer, db, Course, Lecture
+from server.auth.util import hash_password
+from server.tests.base import BaseTestCase
+
+
+class LoginTest(BaseTestCase):
+    def setUp(self):
+        super(LoginTest, self).setUp()
+        self.simon = Lecturer('simon', hash_password('1234'), 'Simon', 'McCallum')
+        db.session.add(self.simon)
+
+        imt3601 = Course('IMT3601 - Game Programming', self.simon)
+        db.session.add(imt3601)
+
+        imt3601_l1 = Lecture('Lecture 1', imt3601)
+        db.session.add(imt3601_l1)
+
+        db.session.flush()
+
+    def test_success(self):
+        res = self.client.post('/auth/login', data={'user_name': 'simon', 'password': '1234'})
```

Comment:
Yes, it will be hashed on client as well. Just think that '1234' is a hash we got from the clientside.